### PR TITLE
Get Range

### PR DESCRIPTION
**Pull request by ayushi2103**

### CODE REVIEW

File: Sources/ETCD/ETCDClient.swift [hunk 1]
```diff
@@ -66,12 +66,13 @@ public final class EtcdClient: @unchecked Sendable {
...
     ///
     /// - Parameter key: The key to fetch the value for. Parameter is of type Sequence<UInt8>.
+    /// - Parameter endKey: Optional. endKey is the key following the last key to delete for the range [key, endKey). If endKey is not given, the range is defined to contain only the key argument. If endKey is one bit larger than the given key, then the range is all the keys with the prefix (the given key). If endKey is '\0', the range is all keys greater than or equal to the key argument
     /// - Returns: A `Value` containing the fetched value, or `nil` if no value was found.
-    public func get(_ key: some Sequence<UInt8>) async throws -> Data? {
-        var rangeRequest = Etcdserverpb_RangeRequest()
-        rangeRequest.key = Data(key)
-
-        let call = client.range(rangeRequest)
+    public func get(_ key: Data, endKey: Data? = nil) async throws -> Data? {
```
Comment (FranzBusch) [inline Sources/ETCD/ETCDClient.swift]: Let's just make this one `get` method that takes a `RangeRequest` parameter

File: Sources/ETCD/RangeRequest.swift [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+// See LICENSE.txt for license information
+// See CONTRIBUTORS.txt for the list of swift-etcd-client-gsoc project authors
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+//===----------------------------------------------------------------------===//
+
+import Foundation
+
+struct RangeRequest {
```
Comment (FranzBusch) [inline Sources/ETCD/RangeRequest.swift]: Let's make this `public` including all the fields and provide the init that takes the key and rangeEnd

File: Sources/ETCD/ETCDClient.swift (hunk 1 above)
Comment (ayushi2103) [reply Sources/ETCD/ETCDClient.swift]: Updated!

File: Sources/ETCD/RangeRequest.swift (hunk 2 above)
Comment (ayushi2103) [reply Sources/ETCD/RangeRequest.swift]: Updated!

File: Sources/ETCD/RangeRequest.swift [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+    var revision: Int = 0
+    var sortOrder: SortOrder = .none
+    var sortTarget: SortTarget = .key
+    var serializable: Bool = false
+    var keysOnly: Bool = false
+    var countOnly: Bool = false
+    var minModRevision: Int = 0
+    var maxModRevision: Int = 0
+    var minCreateRevision: Int = 0
+    var maxCreateRevision: Int = 0
```
Comment (FranzBusch) [inline Sources/ETCD/RangeRequest.swift]: All of these should be public

File: Sources/ETCD/ETCDClient.swift [hunk 4]
```diff
@@ -65,13 +65,11 @@ public final class EtcdClient: @unchecked Sendable {
...
     ///
-    /// - Parameter key: The key to fetch the value for. Parameter is of type Sequence<UInt8>.
+    /// - Parameter rangeRequest: The rangeRequst to get
     /// - Returns: A `Value` containing the fetched value, or `nil` if no value was found.
-    public func get(_ key: some Sequence<UInt8>) async throws -> Data? {
-        var rangeRequest = Etcdserverpb_RangeRequest()
-        rangeRequest.key = Data(key)
-
-        let call = client.range(rangeRequest)
+    public func get(rangeRequest: RangeRequest) async throws -> Data? {
```
Comment (FranzBusch) [inline Sources/ETCD/ETCDClient.swift]:
```suggestion
    public func getRange(_ rangeRequest: RangeRequest) async throws -> Data? {
```

File: Sources/ETCD/ETCDClient.swift [hunk 5]
```diff
@@ -65,13 +65,11 @@ public final class EtcdClient: @unchecked Sendable {
 
     /// Fetch the value for a key from the ETCD server.
```
Comment (FranzBusch) [inline Sources/ETCD/ETCDClient.swift]:
```suggestion
    /// Fetches a range from the ETCD server.
```

Comment (ayushi2103) [reply Sources/ETCD/ETCDClient.swift]: Updated!

File: Sources/ETCD/ETCDClient.swift (hunk 4 above)
Comment (ayushi2103) [reply Sources/ETCD/ETCDClient.swift]: Updated!

File: Sources/ETCD/RangeRequest.swift (hunk 3 above)
Comment (ayushi2103) [reply Sources/ETCD/RangeRequest.swift]: Updated!

**Review by FranzBusch — APPROVED** [review]

**Pull request merged**